share/mk/bsd.cheri.mk

```diff
@@ -51,22 +47,33 @@ WANT_CHERI=	none
 .error CHERI_CC is defined to ${CHERI_CC} which does not exist
 .endif
 
-_CHERI_CC=	${CHERI_CC} -g -integrated-as --target=cheri-unknown-freebsd \
-		-msoft-float
+.if !defined(CHERI_CXX)
+.error CHERI is enabled and request, but CHERI_CXX is undefined
```

Comment:
This should warn and attempt to construct `CHERI_CXX` from `CHERI_CC` to avoid the need to churn all our infrastructure.  This should be safe even if CHERI_CC has clang in the directory parts:
```
CHERI_CXX=${CHERI_CC:H}/${CHERI_CC:T:S/clang/clang++/}
```
Choices that would break this are possible, but as a transition aid this should get the job done.  Once you and @bsdjhb have us moved to clang for mips builds, I'm planning to change the infrastructure to use `XCC` etc as `CHERI_CC` and friends.